pkg/crypto/crypto.go

```diff
@@ -66,9 +66,9 @@ func EncryptConfigMap(configMap []byte, keyPath string, noncePath string) (encCo
 
 // GetConfigMapKeys returns the keys used for encrypting the configMap
 func GetConfigMapKeys(keyPath string, noncePath string) (symmKey []byte, nonce []byte, err error) {
-	if symmetricKey != nil && symmKeyNonce != nil {
+	/*if symmetricKey != nil && symmKeyNonce != nil {
```

Comment:
Why is this change required? This check exists to return an existing key and nonce.

Reply:
Correct. But the issue is, these are global variables and passed values are not checked against them. So for example , in decrypt, if wrong key is passed (any non null value), in stead of returning the passed value, global value is returned. So I prefer to have check in stead of just returning. alternate way can be, compare global value and passed value.

Reply:
Please suggest.

Reply:
Throughout the execution of the `rakshctl` same key should be referred, and hence it's intentional the globally set value should be returned if it already exists. 

I can understand the inhibition to use global variables like that, maybe you can refactor that code to use closures but nevertheless, that function should always return the same key during the entire execution of `rakshctl`

Did you try deploying these changes on a system? 

@bpradipt @mkumatag

Reply:
Please allow me to explain this again.
The function, "GetConfigMapKeys" gets called by both encrypt and decrypt functions.  Using global, as you explained above, is justified. 
However, the issue started happening when I passed wrong key value to decrypt function. The key passed is not null , but it's not the same one which was used to encrypt. Unless I removed this part, the return values were fixed, the global variables; not the ones which user passed.
Hence I wanted to get the user passed values being returned and commented this part. Waiting for comments from you all, @harche @bpradipt @mkumatag

Reply:
I executed this as well. Without the commented out code, with wrong key too, decrypt worked.

Reply:
> Please allow me to explain this again.

I don't know what called for this. It's normal for reviewers to seek more information about the proposed changes.

Reply:
> The key passed is not null , but it's not the same one which was used to encrypt. Unless I removed this part, the return values were fixed, the global variables; not the ones which user passed.

This is intended behavior. That function should pass the same key (global) over and over again. Have you deployed your changes in the real cluster and tested the changes?

Reply:
Thanks much for explaining the behavior of global key Harshal.  I will further explore the crypto functionalities and try to figure out correct condition.
Actions Items from my side
1.  correct the err code part for functions wherever applicable.
2.  revert changes for crypto.go for now.
3. fresh code drop to take care of above action items as part of this PR.

Please let me know if you have any other suggestions.

Reply:
@harche : Can you please review latest code and approve? Also if further comments, please let me know.